src/server/mod.rs

```diff
@@ -1 +1,279 @@
+        player_type: PlayerType::Human,
+        name: client_name,
+        capacity: Board::targets(&Board::init()),
+    };
+
+    // start game
+    start(host, client, client_stream);
+
+    println!("\n{}", Yellow.paint("Bye."));
+}
+
+/// Starting the game with given parameters
+fn start(mut host: Player, mut client: Player, mut stream: TcpStream) {
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+    //                    Request initial board configuration from host                          //
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+
+    if host.capacity == 0 {
+        net::send(
+            &mut stream,
+            MessageType::Text("Server is setting its ships, please wait :)".to_string())
+        );
+        println!("Please set your ships:");
+        match ::model::place_ships(&mut host) {
+            Ok(()) => {},
+            Err(_) => {
+                println!("Failed placing ships!");
+                net::send(&mut stream, MessageType::Quit);
+                return
+            }
+        }
+    }
+    model::print_boards(&host);
+
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+    //                    Request initial board configuration from client                        //
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+    net::send(&mut stream, MessageType::RequestBoard);
+
+    loop {
+        let recv: Result<types::MessageType, DeserializeError> =
+        deserialize_from(&mut stream, bincode::SizeLimit::Infinite);
+        match recv {
+            Ok(received) => {
+                match received {
+                    MessageType::Board(vec) => {
+                        client.own_board = vec.clone();
+                        client.capacity = Board::targets(&vec);
+                        break;
+                    },
+                    MessageType::Quit => {
+                        println!("Client closed connection.");
+                        return
+                    },
+                    _ => continue,
+                }
+            },
+            Err(_) => {
+                println!("ERROR board");
+                net::send(&mut stream, MessageType::Quit);
+                stream.shutdown(Shutdown::Both).expect("shutdown call failed");
+                return
+            },
+        };
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+    //                              Choose random start player                                   //
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+    let mut current_player = match choose_player(&host, &client).name == host.name {
+        true => CurrentPlayer::Host,
+        false => CurrentPlayer::Client,
+    };
```

Comment:
You use `choose_player()` only once (here) and ... it's too complicated. What about:

```rust
let mut current_player = if rand::random() {
    CurrentPlayer::Host
} else {
    CurrentPlayer::Client
};
```